Lib/Rabbit4000/Crypto/SHA1.LIB

```diff
@@ -953,26 +953,33 @@ __xmem void sha_copy_and_swap(void __far* dest, const void __far* src, word lwco
 
 _sha_debug __xmem void sha_copy_and_swap(void __far* dest, const void __far* src, word lwcount)
 {
-	#asm
+   #asm
    ld		px,(sp+@sp+dest)		; dest in PX
-	ld		hl,(sp+@sp+lwcount)
+   ld		hl,(sp+@sp+lwcount)
    ld		bc,hl
    ld		py,(sp+@sp+src)	; source in PY
+   ld		de,4
 .loop:
-	ld		hl,(py+2)
+#if _RAB6K
+   ld		jkhl,(py)
+   swap		jkhl
+   ld		(px),jkhl
+#else
+   ld		hl,(py+2)

```

Comment:
I think you could improve the non-6K code here, based on the technique used in `intel()`:

```
	ex		de,hl
	ld		e,b
	ld		d,c
	ld		b,l
	ld		c,h
```

I'm going to push out a separate commit to make use of the `swap` opcode in `intel()` and `intel16()` on the 6000 platform.

Reply:
Unfortunately, the code from the intel function uses bc and de which we are already using for the loop in this case.
This is probably the best we can do:
   ld		jkhl,(py)
   ld		a,h
   ld		h,L
   ld		L,a
   ex       jk, hl
   ld		a,h
   ld		h,L
   ld		L,a
   ld		(px),jkhl

Reply:
The lack of instructions for accessing j and k or jk is a big pain for these type of operations!